src/components/Dashboard/SafeApps/Grid.tsx

```diff
@@ -0,0 +1,133 @@
+import { ReactElement, useMemo } from 'react'
+import styled from 'styled-components'
+import { Button } from '@gnosis.pm/safe-react-components'
+import { generatePath, Link } from 'react-router-dom'
+import Skeleton from '@material-ui/lab/Skeleton/Skeleton'
+
+import { useAppList } from 'src/routes/safe/components/Apps/hooks/appList/useAppList'
+import { GENERIC_APPS_ROUTE } from 'src/routes/routes'
+import Card, { CARD_HEIGHT, CARD_PADDING, CARD_WIDTH } from 'src/components/Dashboard/SafeApps/Card'
+import ExploreIcon from 'src/assets/icons/explore.svg'
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import { getAppsUsageData, rankTrackedSafeApps } from 'src/routes/safe/components/Apps/trackAppUsageCount'
+
+const StyledGrid = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 20px;
+  flex-wrap: wrap;
+`
+
+const SkeletonWrapper = styled.div`
+  border-radius: 8px;
+  overflow: hidden;
+`
+
+const StyledExplorerButton = styled.div`
+  width: 260px;
+  height: 200px;
+  background-color: white;
+  border-radius: 8px;
+  padding: 24px;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  gap: 8px;
+`
+
+const StyledLink = styled(Link)`
+  text-decoration: none;
+
+  > button {
+    width: 200px;
+  }
+`
+
+// Transactions Builder && Wallet connect
+const featuredAppsId = ['29', '11']
+
+const getRandomApps = (nonRankedApps: SafeApp[], size: number) => {
+  const randomIndexes: string[] = []
+  for (let i = 1; randomIndexes.length < size; i++) {
+    const randomAppIndex = Math.floor(Math.random() * nonRankedApps.length).toString()
+    const randomAppId = nonRankedApps[randomAppIndex].id
+
+    // Do not repeat random apps or featured apps
+    if (!randomIndexes.includes(randomAppIndex) && !featuredAppsId.includes(randomAppId)) {
+      randomIndexes.push(randomAppIndex)
+    }
+  }
+
+  const randomSafeApps: SafeApp[] = []
+  randomIndexes.forEach((index) => {
+    randomSafeApps.push(nonRankedApps[index])
+  })
+
+  return randomSafeApps
+}
+
+const Grid = ({ size = 6 }: { size?: number }): ReactElement => {
+  const { allApps, pinnedSafeApps, togglePin, isLoading } = useAppList()
+
+  const displayedApps = useMemo(() => {
+    if (!allApps.length) return []
+    const trackData = getAppsUsageData()
+    const rankedSafeAppIds = rankTrackedSafeApps(trackData)
+
+    const topRankedSafeApps: SafeApp[] = []
+    rankedSafeAppIds.forEach((id) => {
+      const sortedApp = allApps.find((app) => app.id === id)
+      if (sortedApp) topRankedSafeApps.push(sortedApp)
+    })
+
+    const nonRankedApps = allApps.filter((app) => !rankedSafeAppIds.includes(app.id))
```

Comment:
You could create this array in the `forEach()` above. It's the inverse of `topRankedSafeApps`.